cpp-client/deephaven/dhclient/include/public/deephaven/client/utility/misc_types.h

```diff
@@ -20,4 +22,28 @@ using TimePointSpecifier = std::variant<std::chrono::system_clock::time_point, i
  * string.
  */
 using DurationSpecifier = std::variant<std::chrono::nanoseconds, int64_t, std::string>;
+
+/**
+ * Used to identify OnClose callbacks, eg, allowing their removal after addition.
+ */
+struct OnCloseCbId {
+  std::uint32_t id;
+};
+
+/**
+ * An OnClose callback.
+ */
+using OnCloseCb = std::function<void()>;
+
 } // namespace deephaven::client::utility
+
+// specialized std::less for deephaven::client::utility::OnCloseCbId.
+namespace std {
+using OnCloseCbId = deephaven::client::utility::OnCloseCbId;
+template<> struct less<OnCloseCbId>
+{
+  bool operator() (const OnCloseCbId &lhs, const OnCloseCbId &rhs) const {
+    return lhs.id < rhs.id;
+  }
+};
+} // namespace std
```

Comment:
FYI, except in one specialized circumstance involving `std::hash`, it's very often an anti-pattern to open up `namespace std` and add stuff to it.

There's no need to specialize `std::less` just to get comparisons working. Instead, you can delete this `std::less` specialization and add an `operator<` to your struct as indicated above.

This is more general anyway, because it means that `id1 < id2` will also work as expected.